internals/plan/plan.go

```diff
@@ -317,6 +318,90 @@ const (
 	ActionIgnore   ServiceAction = "ignore"
 )
 
+// LogTargets specifies which log targets should receive a service's logs.
+// It is either an explicit list of targets, or one of the below keywords.
+type LogTargets struct {
+	Targets []string
+	Keyword string
+}
+
+const (
+	// Valid keywords for LogTargets
+	DefaultLogTargets = "default"
+	AllLogTargets     = "all"
+	NoLogTargets      = "none"
+)
+
+func (t *LogTargets) UnmarshalYAML(value *yaml.Node) error {
+	if value.Kind == yaml.ScalarNode {
+		// decode keyword
+		switch value.Value {
+		case DefaultLogTargets, AllLogTargets, NoLogTargets:
+			t.Keyword = value.Value
+			return nil
+		default:
+			return fmt.Errorf("invalid value %q for LogTargets", value.Value)
+		}
+	}
+
+	// unmarshal to []string
+	return value.Decode(&t.Targets)
+}
+
+// Copy returns a deep copy of this LogTargets struct.
+func (t *LogTargets) Copy() *LogTargets {
+	return &LogTargets{
+		Targets: append([]string(nil), t.Targets...),
+		Keyword: t.Keyword,
+	}
+}
+
+// Merge merges the fields from other into t.
+func (t *LogTargets) Merge(other *LogTargets) {
+	if other.Keyword != "" {
+		// anything <- keyword yields keyword
+		t.Keyword = other.Keyword
+		t.Targets = nil
+
```

Comment:
Nits: extraneous blank line (here and below). Also might be cleaner to do an early `return` here and below to avoid the additional `else` and indentation (early returns to avoid nesting are typical in Go).